### PR TITLE
Searching Repo's With Null

### PR DESCRIPTION
**Pull request by SarveshChavan**
Issue 65
Searching Repo's With Null Field

## Changes made
Used **trim()** function to remove leading and trailing spaces

## How does the solution address the problem
This PR will use the **trim()** function and after trimming the content in the search bar it will use the null check and if the null check is passed only then the repos will be searched.

### CODE REVIEW

**Comment by dk-a-dev** [comment]
I found one more exception if u pass any special character 
" TypeErrorImpl was thrown building NewRepoCard(dirty):
Expected a value of type 'int', but got one of type 'String'
@SarveshChavan fix it

**Comment by SarveshChavan** [comment]
I'll work on it.

**Comment by SarveshChavan** [comment]
check the latest commit I followed the following approach.
GitHub allows this set of characters for a repository name [A-Za-z0-9_.-],
So to follow the rules we can use 
inputFormatters: <TextInputFormatter>[
                FilteringTextInputFormatter.allow(RegExp('[ 0-9a-zA-Z._-]',)),
],
a text field property to limit the scope of input characters. We can only allow this set of characters to be included in the text field [ 0-9a-zA-Z._-]
If you are happy with this solution check the PR and let me know if any change required

**Comment by dk-a-dev** [comment]
> check the latest commit I followed the following approach. GitHub allows this set of characters for a repository name [A-Za-z0-9_.-], So to follow the rules we can use inputFormatters: [ FilteringTextInputFormatter.allow(RegExp('[ 0-9a-zA-Z._-]',)), ], a text field property to limit the scope of input characters. We can only allow this set of characters to be included in the text field [ 0-9a-zA-Z._-] If you are happy with this solution check the PR and let me know if any change required

Perfect I was thinking same to put up regex good job buddy its under review, will merge it asap after review

File: lib/screens/search_repos.dart [hunk 1]
```diff
@@ -119,14 +122,15 @@ class _SearchState extends State<Search> {
...
                   FocusScope.of(context).unfocus();
                   setState(() {
-                    setState(() {
-                      listOfRepos = getRepos(_controller.text);
-                    });
+                    listOfRepos = getRepos(_controller.text);
                   });
                 } else {
+                  setState(() {
+                    _controller.text='';
```
Comment (dk-a-dev) [inline lib/screens/search_repos.dart:132]: > _controller.clear()
is preferred over current approach line 132

**Pull request merged**